### DISCUSSION TITLE
Cannot make soft_break work

### DISCUSSION THREAD

**Discussion by pherrymason** [thephpleague/commonmark#653] [Q&A] [answered]
I'm trying to make this text to add a `<br>` line just before the `soft break` sentence, but I have no luck with that.

This is the text I'm trying to convert:

```
This is markdown
-----------------
**bold** *and italics* text
soft break

- one
- two
- three 
- four
```

I've tried using the default options and both forcing it:

```
$converter = new CommonMarkConverter();
return $converter->convertToHtml($html);
```

It gets converted to this:
```
<h2>This is markdown</h2>
<p><strong>bold</strong> <em>and italics</em> text
soft break</p>
<ul>
<li>one</li>
<li>two</li>
<li>three</li>
<li>four</li>
</ul>
```

Maybe I didn't understand properly the meaning of `soft_break`?

**Comment by colinodell** [comment, marked as answer]
[Per the CommonMark spec](http://spec.commonmark.org/0.28/#soft-line-breaks), soft breaks do not render as `<br>` tags by default. 

This library does have a configuration option you can change to `'<br>'` to get that functionality: https://commonmark.thephpleague.com/configuration/

Try making using this instead:

```php
$converter = new CommonMarkConverter([
    'renderer' => [
        'soft_break' => '<br>',
    ]
]);
return $converter->convertToHtml($html);
```

**Comment by pherrymason** [comment]
oh ok, now I understand how to use it. 
Thanks!
